File: chapter01/01/Prog.java
// javac Prog.java
// java Prog

public class Prog {
  public static void main(String[] args) {
    System.out.println("id(1) = " + id(1));
    System.out.println("id(life) = " + id("life"));
  }

  public static <T> T id(T x) {
    return x;
  }
}

File: chapter01/01/prog.hs
-- ghc prog.hs -o a.out
-- ./a.out

import Prelude hiding (id)

id :: a -> a
id x = x

main :: IO ()
main = do putStrLn ("id(1) = " ++ show (id 1))
          putStrLn ("id(life) = " ++ id "life")